### PR TITLE
Batch of expect file removals

### PR DESCRIPTION
**Pull request by eellison**
Another batch of removing expect files. 

One note - I removed the Batched expect files without adding equivalent tests since they are already being tested in another ways, and we are no longer actively maintaining that project.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -11513,7 +11504,9 @@ def f(x, y):
         y = torch.randn(4, 4, dtype=torch.float, device='cuda')
 
         ge = self.checkTrace(f, (x, y))
-        self.assertExpectedGraph(ge.graph_for(x, y))
+        graph = ge.graph_for(x, y)
+        FileCheck().check("broadcast_tensors").check('FusionGroup_0') \
```
Comment (zou3519) [inline test/test_jit.py]: For test_chunk_distributes_cuda:

What we want to check is the following:
(1) broadcast_tensors happens before FusionGroup_0 gets called
(2) There are 2 prim::ConstantChunk inside the FusionGroup_0 definition

If I'm understanding this correctly, this FileCheck expression checks (1) but doesn't explicitly check (2): it would also match the following
```
blah = aten::broadcast_tensors(%2)	
blah = FusionGroup_0(%5, %4)
blah = ConstantChunk(...)
blah = ConstantChunk(...)

with prim::FusionGroup_0 = graph(
...
)
```

I think you can peek into the FusionGroup graph by using node.graph or node.subgraph?

File: test/test_jit.py [hunk 2]
```diff
@@ -11650,7 +11643,9 @@ def foo(hx, cx):
             return torch.cat((hx + cx, hx * cx))
 
         ge = self.checkTrace(foo, (hx, cx))
-        self.assertExpectedGraph(ge.graph_for(hx, cx))
+        graph = ge.graph_for(hx, cx)
+        self.assertAllFused(graph)
+        FileCheck().check("FusedConcat").run(str(graph))
```
Comment (zou3519) [inline test/test_jit.py]: Ideally we'd also check that FusedConcat is the last thing in the FusionGroup

File: test/test_jit.py [hunk 3]
```diff
@@ -11877,18 +11874,23 @@ def test_lstm_cuda(self):
         self.assertGraphContainsExactly(
             forward_graph, 'prim::FusionGroup', 1, consider_subgraphs=True)
         self.assertExpectedGraph(forward_graph, subname='forward')
+        self.assertTrue(len(forward_graph.nodes()) == 2)
+        FileCheck().check("DifferentiableGraph").check_next("TupleConstruct") \
```
Comment (zou3519) [inline test/test_jit.py]: It's a little hard to read this line. I think what we're checking here is that LSTM forward is completely differentiable, except for TupleConstruct. Maybe add a comment? (maybe we should make TupleConstruct differentiable, but that's another discussion).

File: test/test_jit.py [hunk 4]
```diff
@@ -11877,18 +11874,23 @@ def test_lstm_cuda(self):
...
+            .check_next("return").run(str(forward_graph))
 
         hy, cy = module(*inputs)
         (hy + cy).sum().backward()
-        self.assertExpectedGraph(backward_graph(module), subname='backward')
+        backward = backward_graph(module)
+        FileCheck().check("FusionGroup_0").check_next("FusionGroup_1").run(str(backward))
```
Comment (zou3519) [inline test/test_jit.py]: Can we check that there are no more than 2 fusion groups in the backward? The fewer the merrier

Comment (zou3519) [inline test/test_jit.py]: The automated checking is really nice for asserting properties of the forward/backward graphs. In addition to those checks, could we still keep around the forward / backward expect files for this lstm? They're pretty useful as documentation and help debugging and figuring out exactly what is going on with the graphs.

File: test/test_jit.py [hunk 5]
```diff
@@ -11921,7 +11923,8 @@ def cell(x, hx, cx, w_ih, w_hh, b_ih, b_hh):
     def test_lstm_traced_cuda(self):
         inputs = get_lstm_inputs('cuda')
         ge = self.checkTrace(LSTMCellF, inputs)
-        self.assertExpectedGraph(ge.graph_for(*inputs))
+        graph = ge.graph_for(*inputs)
+        FileCheck().check("FusionGroup").run(str(graph))
```
Comment (zou3519) [inline test/test_jit.py]: Would be good to check:
- there is exactly one fusion group
- there are no chunks outside the fusion group
- there are no fusible operations outside the fusion group

File: test/test_jit.py [hunk 6]
```diff
@@ -11930,7 +11933,8 @@ def test_lstm_traced_cpu(self):
         inputs = get_lstm_inputs('cpu')
         try:
             ge = self.checkTrace(LSTMCellF, inputs)
-            self.assertExpectedGraph(ge.graph_for(*inputs))
+            graph = ge.graph_for(*inputs)
+            FileCheck().check("FusionGroup_0[device=-1]").run(str(graph))
```
Comment (zou3519) [inline test/test_jit.py]: I think this is dead code. FusionGroup nodes do not have a device attribute anymore

File: test/test_jit.py [hunk 7]
```diff
@@ -11950,10 +11954,10 @@ def test_milstm_cuda(self):
         self.assertGraphContainsExactly(
             forward_graph, 'prim::FusionGroup', 1, consider_subgraphs=True)
         self.assertExpectedGraph(forward_graph, subname='forward')
-
+        FileCheck().check("DifferentiableGraph").check_next("TupleConstruct") \
+            .check_next("return").check("FusionGroup").run(str(forward_graph))
```
Comment (zou3519) [inline test/test_jit.py]: It would be nice if we could keep around the milstm forward/backward expect files as documentation. It's not as important as the lstm forward/backward expect files though

**Review by zou3519 — COMMENTED** [review]
Went through the fuser tests and left some comments. I'll go through the non-fuser tests later

File: test/test_jit.py (hunk 4 above)
Comment (eellison) [reply test/test_jit.py]: We don't want to test against the expect files anymore. Maybe we could do something like self.updateExpectFile(str(graph)) that doesn't fail if there's a difference ?

File: test/test_jit.py (hunk 5 above)
Comment (eellison) [reply test/test_jit.py]: what do you mean exactly by there is no fusible operations outside the fusion group?

Comment (zou3519) [reply test/test_jit.py]: For LSTM forward, check that there aren't add / sigmoid / tanh operations that are outside the fusion group

File: test/test_jit.py (hunk 4 above)
Comment (zou3519) [reply test/test_jit.py]: Yes, it would be fine to not fail on difference. `self.updateExpectFile(str(graph))` is still going to race, though :(

File: test/test_jit.py [hunk 8]
```diff
@@ -1362,7 +1366,7 @@ def f(x):
             out.copy_(x)
```
Comment (zou3519) [inline test/test_jit.py]: (line 1350) You deleted test_alexnet.expect but it doesn't look like you updated the test. Is the test dead?

Comment (zou3519) [reply test/test_jit.py]: Ditto for test_batchnorm and test_conv, what am I missing?

File: test/test_jit.py [hunk 9]
```diff
@@ -1621,7 +1625,8 @@ def addmm(mat, mat1, mat2, alpha, beta):
         self.run_pass('canonicalize_ops', addmm.graph)
         out_test = addmm(mat, mat1, mat2, alpha, beta)
         self.assertEqual(out_ref, out_test)
-        self.assertExpected(canonical(addmm.graph))
+        # first two addmm replaced
```
Comment (zou3519) [inline test/test_jit.py]: Might be better to separate test_decompose_addmm into two tests: one for the cases that decompose, and one for the cases that don't decompose. This way we can avoid the comment and the checking logic is clearer

File: test/test_jit.py [hunk 10]
```diff
@@ -684,8 +686,9 @@ def f(x, y):
             return x.type_as(y)
 
         tf = torch.jit.trace(f, (a, b))
+        FileCheck().check("type_as").run(str(tf.graph))
         self.run_pass('peephole', tf.graph)
-        self.assertExpectedGraph(tf.graph)
+        FileCheck().check_not("type_as").run(str(tf.graph))
         tf2 = torch.jit.trace(f, (a, c))
         s = str(tf2.graph)
         self.run_pass('peephole', tf2.graph)
```
Comment (zou3519) [inline test/test_jit.py]: (line 692) What is it doing? Can we get rid of it?

Comment (zou3519) [reply test/test_jit.py]: Oh I see, I understand now. Please ignore this comment

File: test/test_jit.py (hunk 8 above)
Comment (eellison) [reply test/test_jit.py]: The tests were updated in a different PR & i forgot to remove the expect - will rebase PR in a sec so it makes sense

File: test/test_jit.py (hunk 4 above)
Comment (eellison) [reply test/test_jit.py]: It will still race but I'm not sure of how to keep around expect files without racing... at least this won't fail master

File: test/test_jit.py [hunk 11]
```diff
@@ -11981,19 +11984,25 @@ def test_lstm_cuda(self):
...
+        # Everything is differentiable but TupleConstruct return
+        FileCheck().check("DifferentiableGraph").check_next("TupleConstruct") \
+            .check_next("return").run(str(forward_graph))
 
         hy, cy = module(*inputs)
         (hy + cy).sum().backward()
-        self.assertExpectedGraph(backward_graph(module), subname='backward')
+        backward = backward_graph(module)
+        FileCheck().check("FusionGroup_0").check_next("FusionGroup_1") \
+            .check_not("FusionGroup_2").run(str(backward))
```
Comment (wanchaol) [inline test/test_jit.py]: I also think it would be beneficial for us to keep the LSTM forward/backward expect file as documentation at least in this intermediate period as we do want to easy to examine the whole graph structure for fusion, that means no assertion should be good enough for us temporarily?

File: test/expect/TestFuser.test_lstm_cuda-backward.expect [hunk 12]
```diff
@@ -1,111 +0,0 @@
-graph(%0 : Float(*, *),
```
Comment (wanchaol) [inline test/expect/TestFuser.test_lstm_cuda-backward.expect]: let's keep this expect file and the forward expect file for now by not asserting failures?

File: test/test_jit.py [hunk 13]
```diff
@@ -2571,8 +2582,7 @@ def single_if(a, b):
         self.assertEqual(res, res_batch.examples())
 
         script_if = torch.jit.script(single_if)
-        graph = torch.to_batch_graph(script_if.graph)
-        self.assertExpected(canonical(graph))
+        torch.to_batch_graph(script_if.graph)
```
[the remark this answers is not in the document]
Comment (eellison) [reply test/test_jit.py]: it still tests that it doesn't fail

**Review by wanchaol — APPROVED** [review]
lgtm, note that there's a file conflict
